readthedocs/core/views/serve.py

```diff
@@ -218,26 +218,26 @@ def _serve_symlink_docs(request, project, privacy_level, filename=''):
 
 
 @map_project_slug
-@map_subproject_slug
-def robots_txt(request, project, subproject, lang_slug=None, version_slug=None):
+def robots_txt(request, project):
     default_robots_fullpath = os.path.join(settings.MEDIA_ROOT, 'robots.txt')
-
-    if not version_slug:
-        version_slug = project.get_default_version()
+    version_slug = project.get_default_version()
 
     filename = resolve_path(
-        subproject or project,  # Resolve the subproject if it exists
+        project,
         version_slug=version_slug,
-        language=lang_slug,
         filename='robots.txt',
         subdomain=True,  # subdomain will make it a "full" path without a URL prefix
     )
     # This breaks path joining, by ignoring the root when given an "absolute" path
     if filename[0] == '/':
         filename = filename[1:]
 
-    public_symlink = PublicSymlink(project)
-    basepath = public_symlink.project_root
+    privacy_level = project.privacy_level
+    if (settings.DEBUG or constants.PUBLIC in serve_docs) and privacy_level != constants.PRIVATE:  # yapf: disable  # noqa
+        symlink = PublicSymlink(project)
+    if (settings.DEBUG or constants.PRIVATE in serve_docs) and privacy_level == constants.PRIVATE:  # yapf: disable  # noqa
+        symlink = PrivateSymlink(project)
+    basepath = symlink.project_root
```

Comment:
Will this file ever exist? I feel like we should be finding it from the default version's HTML root, not from the `project_root`.

Reply:
The default version is appended by the `resolve_path`.

At this point, `filename` is `/en/latest/robots.txt` in my case. Then, I remove the initial `/` and join with `project_root` which ends being `/home/humitos/rtfd/code/readthedocs.org/public_web_root/test-builds/en/latest/robots.txt` (in my local instance) and that file does exist.

Reply:
Ah, I see. 👍